src/engine/guidance/collapse_scenario_detection.cpp

```diff
@@ -61,6 +61,17 @@ bool isShortAndUndisturbed(const RouteStep &step)
     return is_short && noIntermediaryIntersections(step);
 }
 
+bool isValidUturnCombination(const RouteStepIterator first, const RouteStepIterator second)
+{
+    auto has_uturn = hasModifier(*first, DirectionModifier::UTurn) ||
+                     hasModifier(*second, DirectionModifier::UTurn);
+    auto const from_name_change_into_uturn =
+        hasTurnType(*first, TurnType::NewName) && hasModifier(*second, DirectionModifier::UTurn);
+    auto const uturn_into_name_change =
+        hasTurnType(*second, TurnType::NewName) && hasModifier(*first, DirectionModifier::UTurn);
+    return !has_uturn || from_name_change_into_uturn || uturn_into_name_change;
```

Comment:
What do we actually check here? What is the semantic of a "valid uturn combination"? Why is not having uturn modifiers a valid uturn combination for example? And why does it have to include name changes?

My impression was we only want to check if two steps are uturns and then _not_ collapse them. Care to explain why this is not the case?